src/engraving/libmscore/slur.cpp

```diff
@@ -377,17 +377,50 @@ void SlurSegment::avoidCollisions(PointF& pp1, PointF& p2, PointF& p3, PointF& p
 {
     ChordRest* startCR = slur()->startCR();
     ChordRest* endCR = slur()->endCR();
-    Segment* startSeg = startCR ? startCR->segment() : nullptr;
-    Segment* endSeg = endCR ? endCR->segment() : nullptr;
     bool isCrossStaff = (startCR && endCR) ? startCR->vStaffIdx() != endCR->vStaffIdx() : false;
 
+    Segment* startSeg = nullptr;
+    if (isSingleBeginType()) {
+        startSeg = startCR ? startCR->segment() : nullptr; // first of the slur
+    } else {
+        startSeg = system()->firstMeasure()->findFirstR(SegmentType::ChordRest, Fraction(0, 0)); // first of the system
+    }
+    Segment* endSeg = nullptr;
+    if (isSingleEndType()) {
+        endSeg = endCR ? endCR->segment() : nullptr; // last of the slur
+    } else {
+        endSeg = system()->lastMeasure()->last(); // last of the system
+    }
+    if (!startSeg || !endSeg || startSeg == endSeg) {
+        return;
+    }
+
+    // Collect all the segments spanned by this slur segment in a single vector
+    std::vector<Segment*> segList;
+    Segment* seg = startSeg;
+    while (seg && seg->tick() <= endSeg->tick()) {
+        if (seg != startCR->segment()) { // don't include first segment
+            segList.push_back(seg);
+        }
+        if (seg->next() && !seg->next()->isEndBarLineType()) {
+            seg = seg->next();
+        } else if (seg->measure()->next() && seg->measure()->next()->isMeasure()) {
+            seg = toMeasure(seg->measure()->next())->first();
+        } else {
+            break;
+        }
+    }
+    if (segList.empty()) {
+        return;
+    }
+
     // Collision clearance at the center of the slur
     double clearance = 0.75 * spatium(); // TODO: style
     // balance: determines how much endpoint adjustment VS shape adjustment
     // we will do (0 = only shape, 1 = only end point)
     double balance = 0.4; // TODO: style (maybe?)
 
-    if (startSeg && endSeg && startSeg != endSeg && autoplace()) {
+    if (autoplace()) {
```

Comment:
I just realize that this `if` statement can be replaced with 
```c++
if (!autoplace()) {
    return;
}
```
at the very top of the function. That further reduces the number of nested { } blocks. 

Better to do this in a separate commit though, since it creates a lot of changed lines where only the indent level has really changed.

Reply:
Actually, if autoplace is off it doesn't even make sense to call this method, so I can move this check when the method is called.